frontend/src/bundles/common/components/button/button.tsx

```diff
@@ -1,15 +1,68 @@
-type Properties = {
+import { type ReactNode } from 'react';
+
+import { getValidClassNames } from '../../helpers/get-valid-class-names.js';
+
+type ButtonVariant = 'primary' | 'secondary' | 'tertiary';
+type ButtonSize = 'small' | 'medium';
+type ButtonType = 'button' | 'submit';
+
+type ButtonProperties = {
+    size: ButtonSize;
+    variant: ButtonVariant;
     label: string;
-    type?: 'button' | 'submit';
+    leftIcon?: ReactNode;
+    rightIcon?: ReactNode;
+    type?: ButtonType;
+    isDisabled?: boolean;
+    className?: string;
+    onClick?: () => void;
 };
 
-const Button: React.FC<Properties> = ({ type = 'button', label }) => (
-    <button
-        type={type}
-        className="rounded bg-blue-500 px-4 py-2 font-bold text-white hover:bg-blue-700"
-    >
-        {label}
-    </button>
-);
+const classes = {
+    base: 'flex justify-center items-center transition ease-in-out duration-300',
+    size: {
+        small: 'px-4 py-2 h-[32px] text-[14px] font-bold leading-[1.3rem] gap-[4px]',
+        medium: 'px-6 py-4 h-[54px] text-[16px] font-bold leading-[1.3rem] gap-[8px]',
+    },
+
+    variant: {
+        primary:
```

Comment:
It might be a worth to switch the primary and tertiary variants. The yellow button seems to be the primary one to me.